src/lara-plugin/plugin-api/types.ts

```diff
@@ -115,6 +116,19 @@ export interface IEmbeddableRuntimeContext {
   onInteractiveAvailable: (handler: IInteractiveAvailableEventHandler) => void;
   /** True if the interactive is immediately available */
   interactiveAvailable: boolean;
+  /****************************************************************************
+   Function that subscribes provided handler to event that gets called when the interactive's supported features are
+   are available.
+
+   @param handler Event handler function.
+   ****************************************************************************/
+  onInteractiveSupportedFeatures: (handler: IInteractiveSupportedFeaturesEventHandler) => void;
```

Comment:
Note that the PT story says:
>This [EmbeddableRuntimeContext](https://github.com/concord-consortium/lara/blob/master/docs/lara-plugin-api/interfaces/iembeddableruntimecontext.md) Has methods for getting the interactive state. It is to this context that a few new methods likely need to be added.
>1. `getSupportedFeatures()` -- like `getInteractiveState()` this should return a promise or null. Null if the embeddable isn't an interactive. The promise should resolve to the supported features of the interactive once LARA knows what they are.
>2. `sendCustomMessage(type:string, content: object)` -- this should send the message to the wrapped interactive.

As I got into it, however, it seems to me that supported features are more like the interactive available message in that they are set once during interactive initialization and don't change again, so there's no need for a getter that can be called multiple times. Instead it seems more natural to treat it like the interactive available message and have a single callback that is called when appropriate.

Reply:
Supported features should work that way: called once at initialization.
However it has been abused to dynamically change the supported aspect ratio. So in practice an interactive might send the supported features message multiple times.
Since we are adding this new property of supported features, we can require that the first supportedFeatures message includes the available messages. To be safe I think we should also require that the interactive include the available messages property in each supportedFeatures message it sends. 

I'm not sure if this changes how you are implementing this in the plugin-api.

Reply:
>Supported features should work that way: called once at initialization.
However it has been abused to dynamically change the supported aspect ratio. So in practice an interactive might send the supported features message multiple times.

This has been said before, but clearly we should just add a `setAspectRatio()` and possibly a `useAspectRatio()` hook so we can get out of this situation.

>To be safe I think we should also require that the interactive include the available messages property in each supportedFeatures message it sends.

Since most/many interactives won't support any custom messages, it seems like we can still allow an empty custom message specification. The current implementation just ignores it if it's not present.